src/app/app.component.ts

```diff
@@ -37,10 +50,60 @@ export class AppComponent {
         let item: any = part.split('=');
         result[item[0]] = decodeURIComponent(item[1]);
       });
-      if(result['token_json']) {
+      if (result['token_json']) {
         this.authService.logIn(result['token_json']);
       }
     });
+    this.broadcaster.on<Notification>('toast').subscribe(val => {
+      // Trim the list
+      if (this.notificationService.getNotifications.length > AppComponent.MAX_TOAST_NOTIFICATIONS) {
+        for (let i: number = this.notificationService.getNotifications().length - 1; i >= 0; i--) {
+          if (i >= 8) {
+            this.notificationService.remove(this.notificationService.getNotifications()[i]);
+          }
+        }
+      }
+      if (val.primaryAction) {
+        val.primaryAction.id = uuid();
+      }
+      if (val.moreActions) {
+        val.moreActions.forEach(action => action.id = uuid());
+      }
+      this.notificationService.message(
+        val.type.cssClass,
+        val.header,
+        val.message,
+        false,
+        val.primaryAction,
+        val.moreActions
+      );
+    });
+  }
+
+  handleClose($event: any): void {
+    this.notificationService.remove($event.notification);
+    if ($event.notification) {
+      if ($event.notification.primaryAction) {
+        let action = $event.notification.primaryAction as Action;
+        if (action.id) {
+          this._callbacks.delete(action.id);
+        }
+      }
+      if ($event.notification.moreActions) {
+        let moreActions = $event.notification.moreActions as Action[];
+        moreActions.forEach(action => {
+          if (action.id) {
+            this._callbacks.delete(action.id);
+          }
+        });
+      }
+    }
+  }
+
+  handleAction($event: any): void {
```

Comment:
Also, I'm not sure about the sanity of importing ngx-widgets in to libraries like ngx-fabric8-wit which are non-UI libraries. I kind of wanted to abstract it a bit for people.